framework/src/vectorpostprocessors/WorkBalance.C

```diff
@@ -47,18 +47,24 @@ validParams<WorkBalance>()
 WorkBalance::WorkBalance(const InputParameters & parameters)
   : GeneralVectorPostprocessor(parameters),
     _system(getParam<MooseEnum>("system")),
+    _rank_map(_app.rankMap()),
```

Comment:
Is there no way to design this with composition or inheritance? I'd prefer not to have to add a VPP right in the core of MOOSE when most people probably won't use it.

Reply:
Perhaps you could make an unique_ptr to RankMap instead. Then when it's requested through the accessor you can build it? That would avoid having that object active for every simulation on the planet.